docs/AllBridgeFacet.md

```diff
@@ -0,0 +1,82 @@
+# AllBridge Bridge Facet
+
+## How it works
+
+The AllBridge bridge facet works by forwarding calls to the AllBridge core contract on the source chain. It is possible to find the core contract addresses and token pools by curling [token-info endpoint](https://core.api.allbridgecoreapi.net/token-info).
+
+One feature that makes AllBridge different from other bridges is that it uses pools on either side of the bridge. This allows them to no having to have canonical representation on the destination chain. Thus for EVM based chain this means that the pool needs to be approved to spend the bridged tokens.
+
+Underneath, AllBridge can use different message protocols to transfer tokens. These are represented in an enum like
+
+```
+/// @title AllBridge Messenger Protocol Enum
+enum MessengerProtocol {
+    None,
+    Allbridge,
+    Wormhole,
+    LayerZero
+}
+```
+
+```mermaid
+graph LR;
+    D{LiFiDiamond}-- DELEGATECALL -->C(AllBridgeFacet);
+    C(AllBridgeFacet) -- CALL --> C(AllBridge Core)
+```
+
+## Public Methods
+
+- `function startBridgeTokensViaBridgeFacet(BridgeData calldata _bridgeData, AllBridgeData calldata _allBridgeData)`
+  - Simply bridges tokens using bridgeFacet
+- `function swapAndStartBridgeTokensViaBridgeFacet(BridgeData memory _bridgeData, LibSwap.SwapData[] calldata _swapData, AllBridgeData calldata _allBridgeData)`
+  - Performs swap(s) before bridging tokens using bridgeFacet
+
+## Bridge Specific Parameters
+
+Some of the methods listed above take a variable labeled `_allBridgeData`.
+
+This data is specific to allBridgefacet and is represented as the following struct type:
+
+```solidity
+/// @notice The struct for the AllBridge data.
+/// @param fees The amount of token to pay the messenger and the bridge
+/// @param recipient The address of the token receiver after bridging.
```

Comment:
recipient and destinationChainId are part of the bridgeData...what is the reasoning for having duplicates in the AllBridgeData?

Reply:
Because `BridgeData.destinationChainId` is uin256 and `AllBridgeData.destinationChainId` is uint8. Does it makes sense to just cast the `BridgeData.destinationChainId` to Uint8? Also, the chainIds are completely different between AllBridge and LiFi

Reply:
I would keep them separate because they are indeed different.

Reply:
OK, makes sense if they differ. I would suggest to call the variable "allBridgeDestinationChain" or so to make it clear that this is a custom chainId and not the same as standard chainIds

Reply:
Isn't it clear from the parent struct?

Reply:
It's just a suggestion...

if it was super clear, I would not have stumbled upon this in the first place ;)

Reply:
haha true, sure, can update it